src/lib.rs

```diff
@@ -114,31 +121,61 @@ impl ToPromiseString for [Promise] {
     }
 }
 
-#[cfg(any(target_os = "bitrig", target_os = "openbsd"))]
+#[cfg(target_os = "openbsd")]
 mod openbsd;
 
-#[cfg(any(target_os = "bitrig", target_os = "openbsd"))]
-pub use openbsd::pledge_with_paths;
+#[cfg(target_os = "openbsd")]
+pub use openbsd::pledge;
 
-#[cfg(not(any(target_os = "bitrig", target_os = "openbsd")))]
-pub fn pledge_with_paths(_: &str, _: &[&std::path::Path]) -> Result<(), Error> {
+#[cfg(not(target_os = "openbsd"))]
+pub fn pledge<'p, 'e, P, E>(_: P, _: E) -> Result<(), Error>
+where P: Into<Option<&'p str>>, E: Into<Option<&'e str>> {
```

Comment:
we could simplify this by using [impl Trait in argument position](https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/impl-trait-for-returning-complex-types-with-ease.html), but the minimum Rust version we want to support was unclear to me

Reply:
The minimum version I was targeting was probably "latest minus 2" or something like that. I'm good with using impl Trait in argument position!